samples/servicecontrol/adapter-sqlserver/sample.md

```diff
@@ -0,0 +1,119 @@
+ 1. Go to the Shipping console and press `f` to simulate message processing failure.
+ 1. Go back to Sales and press `o` to create yet another order. Notice the `OrderAccepted` event fails in Shipping and is moved to the error queue.
+ 1. Press `f` again to disable message processing failure simulation in Shipping.
+ 1. Open ServicePulse and select the Failed Messages view.
+ 1. Notice the existence of one failed message group with two messages. Open the group.
+ 1. One of the messages is `OrderAccepted` which failed in `Shipping`, the other is `OrderShipped` which failed in `Sales`.
+ 1. Press the "Retry all" button.
+ 1. Go to the Shipping console and verify that the `OrderAccepted` event has been successfully processed.
+ 1. Go to the Sales console and verify that both `OrderShipped` events have been successfully processed.
+ 1. Shut down the Shipping endpoint.
+ 1. Open ServicePulse and notice a red label next to the heart icon. Click on the that icon to open the Endpoints Overview. Notice that `Samples.ServiceControl.SqlServerTransportAdapter.Shipping` is now displayed in the Inactive Endpoints tab.
+
+
+## Code walk-through 
+
+The solution consists of four projects.
+
+
+### Shared
+
+The Shared project contains the message contracts and the physical topology definition. The topology is defined in the `Connections` class via a method that takes the name of the queue table ([physical address](/nservicebus/sqlserver/addressing.md)) and returns the connection string to be used to access that queue.
+
+snippet: GetConnectionString
+
+The `StartsWith` comparison ensures that the [satellite](/nservicebus/satellites/) queues are correctly addressed. The `poison` queue is used by the adapter for unrecoverable failures. 
+
+This topology is used in business endpoints (Sales, Shipping) as well as in the Adapter.
+
+
+### Sales and Shipping
+
+The Sales and Shipping projects contain endpoints that simulate execution of business process. The process consists of two events: `OrderAccepted` published by Sales and subscribed by Shipping and `OrderShipped` published by Shipping and subscribed by Sales.
+
+The Sales and Shipping endpoints use separate databases and their transports are configured in the [multi-instance](/nservicebus/sqlserver/deployment-options.md#modes-overview-multi-instance) mode using the topology definition from the `Connections` class.
+
+The business endpoints include message processing failure simulation mode (toggled by pressing `f`) which can be used to generate failed messages for demonstrating message retry functionality.
+
+The Shipping endpoint has the Heartbeats plug-in installed to enable uptime monitoring via ServicePulse.
+
+
+### Adapter
+
+The Adapter project hosts the `ServiceControl.TransportAdapter`. The adapter has two sides: endpoint-facing and ServiceControl-facing. Each side can use different transport but in this sample both use SQL Server transport:
```

Comment:
> Each side can use different transport

Given the purpose of this sample I'd skip that or replace with "ServiceControl can use a different transport internally than endpoints it is monitoring, but in this sample both will use SQL Server transport for simplicity"